tests/queries/0_stateless/02151_http_s_structure_set_eof.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+tmp_file=$(mktemp "$CURDIR/clickhouse.XXXXXX.csv")
+trap 'rm $tmp_file' EXIT
+
+# NOTE: should be huge enough
+$CLICKHOUSE_CLIENT -q "SELECT toString(number) FROM numbers(10e6) FORMAT TSV" > $tmp_file
+
+# Regression for one of the following assertions from ExternalTablesHandler:
+#
+#     - ./src/IO/ReadBuffer.h:58: bool DB::ReadBuffer::next(): Assertion `!hasPendingData()' failed.
+#     - ./src/Server/HTTP/HTMLForm.cpp:245: bool DB::HTMLForm::MultipartReadBuffer::skipToNextBoundary(): Assertion `boundary_hit' failed.
+#     - ./src/IO/LimitReadBuffer.cpp:17: virtual bool DB::LimitReadBuffer::nextImpl(): Assertion `position() >= in->position()' failed.
+#
+# NOTE: Just in case check w/ input_format_parallel_parsing and w/o
+timeout 0.15s ${CLICKHOUSE_CURL} -sS -F "s=@$tmp_file;" "${CLICKHOUSE_URL}&s_structure=key+Int&query=SELECT+dummy+IN+s&input_format_parallel_parsing=true" -o /dev/null
+echo $?
+timeout 0.15s ${CLICKHOUSE_CURL} -sS -F "s=@$tmp_file;" "${CLICKHOUSE_URL}&s_structure=key+Int&query=SELECT+dummy+IN+s&input_format_parallel_parsing=false" -o /dev/null
```

Comment:
If I understand correctly, you prepare a huge `tmp_file` so it can't be uploaded in 0.15s, and then you stop uploading after 0.15s, and thus the server will receive an unexpected EOF. Right?

Reply:
Yes, and the test checks that the command will be timed out (124 exit code).

Reply:
I think we need those two things in the comments in the test.

Reply:
Updated.